### PR TITLE
Add patching objects with tags tests

### PR DESCRIPTION
**Pull request by kyriechen96**
Issue #, if available:
Need to validate that tags could be added together with modification of other fields.

Description of changes:
Add changes of tags together while modifying other fields.
Remove a redundant test step.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/e2e/scenarios/ACL/acl_create_update.yaml [hunk 1]
```diff
@@ -74,12 +74,15 @@ steps:
       spec:
         userNames:
           - userone$RANDOM_SUFFIX
+        tags:
+          - key: "test_key_1"
+            value: "test_value_1"
```
Comment (A-Hilaly) [inline test/e2e/scenarios/ACL/acl_create_update.yaml:79]: Curious here, how are you asserting that these tags got applied to the ACL?

Comment (RedbackThomson) [reply test/e2e/scenarios/ACL/acl_create_update.yaml:79]: I had the same comment in another PR - we aren't validating that the AWS resource actually has the tags.

File: test/e2e/scenarios/ACL/acl_create_update.yaml [hunk 2]
```diff
@@ -74,12 +74,15 @@ steps:
...
     wait:
       status:
         conditions:
           ACK.ResourceSynced:
             status: "True"
-            timeout: 180
+            timeout: 300
```
Comment (RedbackThomson) [inline test/e2e/scenarios/ACL/acl_create_update.yaml:85]: No way it takes 5 minutes to update tags. That's insane!

Comment (kyriechen96) [reply test/e2e/scenarios/ACL/acl_create_update.yaml:85]: With 180 second it failed. I checked controller logs, and logs show updating tags all the time (with different orders) so `ResourceSynced` cannot be true. Should we pause this PR and wait for new tag compare function being added to `code-generator` and `runtime`?

Comment (kyriechen96) [reply test/e2e/scenarios/ACL/acl_create_update.yaml:85]: I've tried locally. It really took a long time to finish update. More than 5 minutes. Really think we need to wait for new version of code-generator/runtime that includes new compare tags function/method. Extra tags update calls slow everything down.

File: test/e2e/scenarios/ACL/acl_create_update.yaml (hunk 1 above)
Comment (kyriechen96) [reply test/e2e/scenarios/ACL/acl_create_update.yaml:79]: Resolved in PR #65

**Review by A-Hilaly — APPROVED** [review]
+++

**Review by RedbackThomson — APPROVED** [review]
/lgtm

**Pull request merged**